core/src/main/java/com/yahoo/oak/NativeMemoryAllocator.java

```diff
@@ -196,13 +196,21 @@ public boolean isClosed() {
 
     // When some buffer need to be read from a random block
     @Override
-    public void readByteBuffer(Slice s) {
+    public void readByteBuffer(Slice s, int blockID) {
+
+        int sliceBlockID = s.getAllocatedBlockID();
+
+        // do we need to update the buffer?
+        if (sliceBlockID == blockID) {
+            return;
+        }
```

Comment:
> do we need to update the buffer?

Yes! We should update the buffer anyway.
This can be a Slice that is from a different allocator.
Hence, the blockID can be the same, but it refers to a different ByteBuffer.

Reply:
I do not think we should update the buffer anyway. Memory manager and allocator are per OakMap instance entities. This is currently an invariant and I do not see it changing soon. I do not think Blocks should be travel/shared among different OakMaps then the entire memory management and version numbers need to be shared and etc. I do not see a good reason to do that all.